agent/api/statechange.go

```diff
@@ -377,3 +459,132 @@ func (ts TaskStateChange) GetEventType() statechange.EventType {
 func (AttachmentStateChange) GetEventType() statechange.EventType {
 	return statechange.AttachmentEvent
 }
+
+func buildManagedAgentStateChangePayload(change ManagedAgentStateChange) *ecsmodel.ManagedAgentStateChange {
+	if !change.Status.ShouldReportToBackend() {
```

Comment:
Q: are there any required fields that we need to validate like we do in `buildContainerStateChangePayload`?

Reply:
Er never mind it looks like ShouldReportToBackend is doing that

Reply:
I don't believe so - currently ManagedAgentStateChange structs are only initialized using the `NewManagedAgentChangeEvent` function defined on L209-L230. Tracing where and how this function is called, we expect that this function and its downstream callers do the necessary validation of a `ManagedAgentStateChange` prior to the point of it being passed into the `buildManagedAgentStateChangePayload` function (i.e., the relevant fields of any given `ManagedAgentStateChange` passed into the `buildManagedAgentStateChangePayload` function should be populated).